### PR TITLE
Travis CI: Upgrade to Python 3.9

### PR DESCRIPTION
**Pull request by cclauss**
Travis CI now supports Python 3.9

https://github.com/scipy/scipy/blob/master/.github/workflows/bleedingedge.yml

### CODE REVIEW

**Comment by dhruvmanila** [comment]
There are dependencies which still haven't upgraded yet. But I will take a look into on what is happening here. Give me some time.

**Comment by dhruvmanila** [comment]
Should we switch to GitHub actions ?

**Comment by cclauss** [comment]
@dhruvmanila Can I please get your review on this one?

**Comment by dhruvmanila** [comment]
Yes, sorry, on it right away.

File: machine_learning/forecasting/run.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def linear_regression_prediction(
     input : training data (date, total_user, total_event) in list of float
     output : list of total user prediction in float
     >>> linear_regression_prediction([2,3,4,5], [5,3,4,6], [3,1,2,4], [2,1], [2,2])
-    5.000000000000003
+    4.9999999999999964
```
Comment (dhruvmanila) [inline machine_learning/forecasting/run.py:29]: This must be a floating-point arithmetic error. Maybe assert the difference? 

`abs(answer - expected) < 0.00001`

**Review by dhruvmanila — COMMENTED** [review]
Weird. Travis is installing `qiskit-aer` https://travis-ci.com/github/TheAlgorithms/Python/jobs/428734520#L372

Even in the `qiskit-aer` source code the attribute is present: https://github.com/Qiskit/qiskit-aer/blob/master/qiskit/providers/aer/__init__.py#L74

Why is it showing an `AttributeError` then?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+#      - gfortran
+#      - libgmp-dev
+#      - libmpfr-dev
+#      - libsuitesparse-dev
+#      - ccache
+#      - libmpc-dev
+env:
+  - PATH=$PATH:$HOME/.cargo/bin
+before_install:
+  - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```
Comment (dhruvmanila) [inline .travis.yml:20]: Why are you installing `rust`?

**Comment by dhruvmanila** [comment]
@cclauss I think this should be closed. If you think otherwise, please feel free to reopen it.

**Pull request closed without merging**